### ISSUE TITLE
Design for accessing spanish version of site

### ISSUE THREAD

**Issue by BethMattern** [usds/justice40-tool#721]

**Comment by janelien** [comment]
Mockups are available under "Header + Footer"

Titled:
1024 header_with Spanish
880-1023 header_with Spanish
480-639 header_with Spanish

**Comment by janelien** [comment]
Reference https://civilrights.justice.gov/ for interaction

**Comment by switzersc-usds** [comment]
Note that another next step here is Vim will generate the latest en.json and send to Katherine for verification that it includes all content and is a suitable format to send to translation team to use for generating an es.json to send back to us. Vim will also check out the code to make sure we're all set to quickly integrate the content once we have it.

**Comment by vim-usds** [comment]
@katherinedm-usds, excited to be starting this effort! A quick primer on internationalization (aka i18n)!

### 1. Validating English internationalization (aka i18n) copy
 Here's the English i18n cop (aka _the en.json file_):
- https://github.com/usds/justice40-tool/blob/main/client/src/intl/en.json

It _should_ have the copy for the entire site. If you find anything that's missing, please let me know. 

A quick aside on the explanation of this en.json file in case it's new for you. Specifically, let's look at [line 318](https://github.com/usds/justice40-tool/blob/c99677f9b3ca4bc2e8819f6f7d357e145d14d726/client/src/intl/en.json#L318) of the en.json file:

```json
{
  header.about: {
    "defaultMessage": "About",
    "description": "Header navigate item to the about page"
  },
  header.contact: {
    "defaultMessage": "Contact",
    "description": "Header navigate item to the Contact page"
  },
  header.explore: {
    "defaultMessage": "Explore the tool",
    "description": "Header navigate item to the Explore the tool page"
  },
  header.methodology: {
    "defaultMessage": "Data & methodology",
    "description": "Header navigate item to the Methodology page"
  },
}
```

This part of the en.json file holds the text of the main navigation links, e.g. _About_, _Contact_, etc. The _keys_ are highlighted in red. These _keys_ (as we shall soon see) is the mapping between English and Spanish.

### 2. Mapping from English to Spanish
An incomplete Spanish i18n copy (aka es.json file) can be seen here with some place holder translations:
- https://github.com/usds/justice40-tool/blob/main/client/src/intl/es.json

Looking at [line 9](https://github.com/usds/justice40-tool/blob/c99677f9b3ca4bc2e8819f6f7d357e145d14d726/client/src/intl/es.json#L9) of the es.json:

```json
{
  header.about: "Sobre nosotras",
  header.contact: "Contacto",
  header.explore: "Explore el trabajo",
  header.methodology: "Metodología",
}
```

The above Spanish content has to use the same keys as the English content. If these match, the website will substitute English for Spanish.

> **Fun fact**: each line in the above es.json file is the famous _key-value pair_. Where `header.about` is the _key_ and `Sobre nosotros` is the _value_.

### 3. Filling out es.json
The es.json is sparse and it's _keys_ may even be outdated. Let's use the keys in en.json as the source of truth to update the es.json keys along with the adding translated values.

If any of this is clear as mud, please let me know!

**Comment by vim-usds** [comment]
Creating a ticket to start working on adding UI to switch between English and Spanish, #747

**Comment by katherinedm-usds** [comment]
Is there anything else that's needed for this issue, or can it be closed, and work tracked in #747 and other issues?

**Comment by katherinedm-usds** [comment]
Vim confirmed this could be closed now, dev has what they need - thx!
